File: solutions/easy/001-two-sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
  const hashmap: Map<number, number> = new Map();

  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];

    if (hashmap.has(complement)) {
      return [hashmap.get(complement)!, i];
    }

    hashmap.set(nums[i], i);
  }

  return [];
}

console.log(twoSum([2, 7, 11, 15], 9));

File: solutions/easy/001-two-sum/solution.py
def twoSum(nums: list[int], target: int) -> list[int]:
    map = {}

    for i in range(len(nums)):
        complement = target - nums[i]

        if complement in map:
            return [map[complement], i]

        map[nums[i]] = i

    return []


print(twoSum([1, 5, 6, 3], 4))
